src/contractTest/java/uk/gov/hmcts/reform/orgrolemapping/OrgRoleMappingConsumerTestForCRD.java

```diff
@@ -0,0 +1,293 @@
+                .method(HttpMethod.POST.toString())
+                .body(createCRDMultipleUsersRequest(),String.valueOf(ContentType.JSON))
+                .willRespondWith()
+                .status(HttpStatus.OK.value())
+                .headers(getResponseHeaders())
+                .body(createCRDMultipleUsersResponse())
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "executeGetCRDProfileUsingFetchByUserIdAndGet200")
+    void getCRDProfileUsingFetchByUserIdAndGet200Test(MockServer mockServer)
+            throws JSONException, IOException {
+        String actualResponseBody =
+                SerenityRest
+                        .given()
+                        .headers(getHttpHeaders())
+                        .contentType(ContentType.JSON)
+                        .body(createCRDUserRequest())
+                        .post(mockServer.getUrl() + CRD_GET_ROLE_ASSIGNMENT_URL)
+                        .then()
+                        .log().all().extract().asString();
+
+        JSONObject jsonResponse = new JSONObject(actualResponseBody);
+        JSONArray crdResponse = (JSONArray)jsonResponse.get("");
+        JSONObject firstRecord = (JSONObject)crdResponse.get(0);
+        JSONArray roleObject = (JSONArray)firstRecord.get("role");
+        JSONObject roleRecord = (JSONObject)roleObject.get(0);
+        assertThat(roleRecord.get("role"), equalTo("senior-tribunal-caseworker"));
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "executeGetCRDProfileMultipleUsersUsingFetchByUserIdAndGet200")
+    void getCRDProfileMultipleUsersUsingFetchByUserIdAndGet200Test(MockServer mockServer)
+            throws JSONException, IOException {
+        String actualResponseBody =
+                SerenityRest
+                        .given()
+                        .headers(getHttpHeaders())
+                        .contentType(ContentType.JSON)
+                        .body(createCRDMultipleUsersRequest())
+                        .post(mockServer.getUrl() + CRD_GET_ROLE_ASSIGNMENT_URL)
+                        .then()
+                        .log().all().extract().asString();
+
+        JSONObject jsonResponse = new JSONObject(actualResponseBody);
+        JSONArray crdResponse = (JSONArray)jsonResponse.get("");
+        JSONObject firstRecord = (JSONObject)crdResponse.get(0);
+        JSONArray roleObject = (JSONArray)firstRecord.get("role");
+        JSONObject roleRecord = (JSONObject)roleObject.get(0);
+        assertThat(roleRecord.get("role"), equalTo("senior-tribunal-caseworker"));
+
+        JSONObject secondRecord = (JSONObject)crdResponse.get(1);
+        JSONArray roleObject2 = (JSONArray)secondRecord.get("work_area");
+        JSONObject roleRecord2 = (JSONObject)roleObject2.get(0);
+        assertThat(roleRecord2.get("service_code"), equalTo("BFA1"));
+    }
+
+    private DslPart createCRDResponse() {
+        return newJsonBody(o -> o
+                .minArrayLike("", 1, 1,
```

Comment:
Provide the name of array here instead of "". As, it will easy to understand and iterate.

Reply:
At the moment, CRD response doesn't have any name for this. Will update once we got the name from CRD team.